src/components/structured-text-block/structured-text-block.vue

```diff
@@ -1,16 +1,31 @@
 <template>
+  <aside class="structured-text__aside">
+    <toc-section
+      v-if="toc"
+      :items="tocItems"
+    />
+  </aside>
   <DatocmsStructuredText
+    v-bind="$attrs"
     :data="content"
     :render-block="renderBlock"
     :custom-node-rules="customNodeRules"
     class="structured-text"
     :class="{
       'structured-text--center-grid': gridAlignment === 'center',
+      'structured-text--with-toc': toc,
     }"
   />
 </template>
 
+<script>
+export default {
+  inheritAttrs: false
+}
+</script>
```

Comment:
It needs to be on a separate script according to [the docs](https://vuejs.org/guide/components/attrs.html#disabling-attribute-inheritance)